packet.go

```diff
@@ -698,7 +699,14 @@ func (p *sshFxpOpenPacket) MarshalBinary() ([]byte, error) {
 	b = marshalUint32(b, p.Pflags)
 	b = marshalUint32(b, p.Flags)
 
-	return b, nil
+	payload := marshal(nil, p.Attrs)
```

Comment:
I kinda made the assumption that the old code is OK and didn't review it at all. This is the same thing that's being done for the stat packets as well. Is there a need to handle open differently?

Reply:
I would probably use:
```
switch attr := p.Attr.(type) {
case []byte:
  return b, payload, nil
default:
  b = marshal(b, p.Attrs) // in this case, we may as well use the append feature of the marshaling.
  return b
}
```

Reply:
Changing this seems like a can of worms I would like to avoid opening. 😅 There seem to exist code-paths for when an unexported `marshalPacket` exists (compared to `MarshalBinary` only, as your suggestion would require), and changing this would mean we diverge from the impl. in `sshFxpSetstatPacket` and `sshFxpFsetstatPacket`.

Reply:
🤔 OK, sounds good. We can likely push it off. When (if? 😭) we get the V2 API working, the `encoding/ssh/filexfer` covers this without relying on `any` types.